common/app-starters-task-launch-request-common/src/test/java/org/springframework/cloud/stream/app/tasklaunchrequest/TaskLaunchRequestIntegrationTests.java

```diff
@@ -0,0 +1,204 @@
+	@Test
+	public void dataflowTaskLaunchRequestWithTaskLaunchRequestContext() throws IOException {
+
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
+			TestChannelBinderConfiguration.getCompleteConfiguration(TestApp.class)).web(WebApplicationType.NONE)
+			.run("--spring.jmx.enabled=false",
+				"---task.launch.request.task-name=foo", "---task.launch.request.args=foo=bar,baz=boo")) {
+
+			TaskLaunchRequestContext taskLaunchRequestContext = new TaskLaunchRequestContext();
+
+			taskLaunchRequestContext.addCommandLineArg("localFile=/some/file/path");
+			taskLaunchRequestContext.addCommandLineArg("process=true");
+
+			DataFlowTaskLaunchRequest dataFlowTaskLaunchRequest = verifyAndreceiveDataFlowTaskLaunchRequest(context,
+				taskLaunchRequestContext);
+
+			assertThat(dataFlowTaskLaunchRequest.getTaskName()).isEqualTo("foo");
+			assertThat(dataFlowTaskLaunchRequest.getCommandlineArguments()).containsExactlyInAnyOrder("foo=bar",
+				"baz=boo", "localFile=/some/file/path", "process=true");
+		}
+	}
+
+	@Test
+	public void dataflowTaskLaunchRequestWithTaskLaunchRequestContextProvider() throws IOException {
+
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
+			TestChannelBinderConfiguration.getCompleteConfiguration(TestApp.class)).web(WebApplicationType.NONE)
+			.run("--spring.jmx.enabled=false",
+				"---task.launch.request.task-name=foo", "--enhanceTLRContext=true")) {
+
+			DataFlowTaskLaunchRequest dataFlowTaskLaunchRequest = verifyAndreceiveDataFlowTaskLaunchRequest(context);
+
+			assertThat(dataFlowTaskLaunchRequest.getTaskName()).isEqualTo("foo");
+			assertThat(dataFlowTaskLaunchRequest.getCommandlineArguments()).hasSize(1);
+			assertThat(dataFlowTaskLaunchRequest.getCommandlineArguments()).containsExactly("runtimeArg");
+		}
+	}
+
+	private DataFlowTaskLaunchRequest verifyAndreceiveDataFlowTaskLaunchRequest(ApplicationContext context)
+		throws IOException {
+		return this.verifyAndreceiveDataFlowTaskLaunchRequest(context, null);
+	}
+
+	private DataFlowTaskLaunchRequest verifyAndreceiveDataFlowTaskLaunchRequest(ApplicationContext context,
+		TaskLaunchRequestContext taskLaunchRequestContext) throws IOException {
+
+		MessageChannel input = context.getBean("input", MessageChannel.class);
+
+		PollableChannel output = context.getBean("output", PollableChannel.class);
+
+		MessageBuilder<byte[]> builder = MessageBuilder.withPayload(new byte[] {});
+
+		if (taskLaunchRequestContext != null) {
+			builder.setHeader(TaskLaunchRequestContext.HEADER_NAME, taskLaunchRequestContext);
+		}
+
+		input.send(builder.build());
+
+		Message<?> message = output.receive(1000);
+
+		assertThat(message).isNotNull();
+
+		return (DataFlowTaskLaunchRequest) message.getPayload();
+	}
+
+	@Configuration
+	@EnableAutoConfiguration
+	static class TestApp {
+		@Bean
+		public TaskLaunchRequestContextProvider taskLaunchRequestContextProvider(@Value("${enhanceTLRContext:false}")
+			boolean enhance) {
```

Comment:
Suspicious wrapping. I think I have similar problem with the latest IDEA. It doesn't wrap smart enough with the related content for the entity.
So, I'd prefer this then:
```suggestion
public TaskLaunchRequestContextProvider taskLaunchRequestContextProvider(
                 @Value("${enhanceTLRContext:false}") boolean enhance) {
```